docs/baremetal-agent-controller.md

```diff
@@ -0,0 +1,205 @@
+At the end of this process you will have the assisted-service running under the `assisted-installer`
+namespace.
+
+Take notice of the [assisted-service-patch-image][aspi-custom] kustomization if you are testing a local
+assisted-service image. More on this in the operator docs. Another, more hacky option, is to
+manually modify the image for the `assisted-service` deployment after it's been deployed. Favor the
+kustomization way over this.
+
+
+```
+[dev@edge-10 dev-scripts]$ oc get pods
+NAME                                READY   STATUS    RESTARTS   AGE
+assisted-service-7d45ccf4f8-88pw4   1/1     Running   0          18h
+ocp-metal-ui-7967c757b4-fdxph       1/1     Running   0          21h
+postgres-6bd4dc768f-54pgz           1/1     Running   0          21h
+```
+
+Tailing the logs for the `assisted-service` pod should not show any errors.
+
+Creating BareMetalHost resources
+===
+
+The [baremetal operator][bmo] creates the `BareMetalHost` resources for the existing nodes
+automatically. For scenarios using extra worker nodes (like SNO), it will be necessary to create
+`BareMetalHost` resources manually. Lucklily enough, `dev-scripts` is one step ahead and it has
+prepared the manifest for us already.
+
+```
+less ocp/ostest/extra_host_manifests.yaml
+```
+
+BMAC will disable inspection for this `BareMetalHost` resource automatically but, if you want to get
+ahead and save some time, you can disable it by modifying the manifest generated by `dev-scripts`
+and adding the following annotation:
+
+```
+  labels:
+    # optionally add the installenvs annotation, if needed for BMAC
+    installenvs.adi.openshift.io: "bmac-test"
+
+  annotations:
+    # BMAC will add this annotation if not present
+    inspect.metal3.io: disabled
+```
+
+The above is an optional step. If skipped, the `BareMetalHost` will boot IPA and spend some time in
+the inspecting phase when the manifest is applied. There's an [open PR](https://github.com/openshift-metal3/dev-scripts/pull/1193/)
+that will allow setting an environment variable to have inspection disabled by `dev-scripts` by default.
+
+
+[bmo]: https://github.com/openshift/baremetal-operator
+[local-bmo]: https://github.com/openshift-metal3/dev-scripts/blob/master/metal3-dev/local-bmo.sh
+[aspi-custom]: https://github.com/openshift/assisted-service/blob/master/config/default/assisted-service-patch-image.yaml
+
+
+Creating ClusterDeployment and InstallEnv
+==
+
+Before deploying the ClusterDeployment, make sure you have created a secret with your pull-secret.
+In this environment it's called `my-pull-secret`
+
+```
+kubectl create secret -n assisted-installer generic my-pull-secret --from-file=.dockerconfigjson=pull_secret.json
+```
+
+```
+apiVersion: hive.openshift.io/v1
+kind: ClusterDeployment
+metadata:
+    name: bmac-test
+    namespace: assisted-installer
+spec:
+  baseDomain: bmac.hungry-bytes.io
+  clusterName: bmac-test
+  platform:
+    agentBareMetal:
+      apiVIP: ""
+      ingressVIP: ""
```

Comment:
Getting `spec.platform.agentBareMetal.agentSelector: Required value`. Needs:

`agentSelector:
        matchLabels:
          bla: aaa`

Reply:
I accidentally put it below in the wrong section :)